packages/hyperion-flowlet/test/FlowletWrapper.test.ts

```diff
@@ -150,8 +150,27 @@ describe("test flow of flowlets", () => {
     } finally {
       expect(getTriggerFlowlet(p2)).toBe(tf);
     }
-    const allSettled = Promise.allSettled([p0, p1, p2]);
-    expect(getTriggerFlowlet(allSettled)?.getFullName()).toMatch(/Promise.allSettled\(\d+&\d+&\d+\)/);
+
+    main.data.triggerFlowlet = new flowletManager.flowletCtor("tf2");
+    const p3 = Promise.resolve(2);
+
+    const all = Promise.all([p0, p1, p3]);
+    expect(getTriggerFlowlet(all)?.getFullName()).toMatch(/Promise.all\(\d+&\d+\)/); // p0 and p1 have the same TF, P3 has a new one
+
+    const race = Promise.race([p0, p1, p3]);
+    expect(getTriggerFlowlet(race)?.getFullName()).toMatch(/Promise.race\(\d+&\d+\)/);
+
+    const allSettled = Promise.allSettled([p0, p1, p2, p3]);
+    expect(getTriggerFlowlet(allSettled)?.getFullName()).toMatch(/Promise.allSettled\(\d+&\d+\)/);
```

Comment:
Noob question: I wonder why we had 3 TF before, but now getting only 2.

Reply:
p0, p1, and p2 have the same trigger flowlet, and this PR removes repeated trigger flowlet values